### PR TITLE
feat(generator): add SYSLOG_HOST context variable for syslog forwarding

### PR DESCRIPTION
**Pull request by duplico**
## Summary

- Adds \`SYSLOG_HOST\` context variable support to vyos-onecontext
- New \`SyslogGenerator\` in \`generators/system.py\` emits \`set system syslog host <SYSLOG_HOST> facility all level info\`
- Parser reads \`SYSLOG_HOST\` from OpenNebula context and passes it through \`RouterConfig\`
- \`validate_syslog_host\` field validator rejects shell-injection vectors (whitespace, quotes, shell metacharacters); empty/whitespace-only values become \`None\`
- Feature is opt-in: if \`SYSLOG_HOST\` is not set, no syslog configuration is emitted
- \`syslog_host\` field and validator placed together after \`validate_ssh_key\`, following file conventions

**Flow:** VyOS router → syslog UDP 514 → Alloy (monitoring VM) → Loki

Closes #3170

## Test plan

- [x] \`TestSyslogGenerator::test_generate_with_hostname\` - emits correct command with FQDN host
- [x] \`TestSyslogGenerator::test_generate_with_ip_address\` - emits correct command with IPv4
- [x] \`TestSyslogGenerator::test_generate_with_ipv6_bare\` - emits correct command with bare IPv6
- [x] \`TestSyslogGenerator::test_generate_with_ipv6_bracketed\` - emits correct command with bracketed IPv6
- [x] \`TestSyslogGenerator::test_generate_without_syslog_host\` - returns empty list when None
- [x] \`TestSyslogGenerator::test_generate_config_includes_syslog\` - \`generate_config()\` includes syslog command
- [x] \`TestSyslogGenerator::test_generate_config_no_syslog_when_unset\` - \`generate_config()\` omits syslog when unset
- [x] \`TestInputValidation::test_syslog_host_*\` - 13 validator tests covering rejection (injection, space, newline, pipe, quote), acceptance (IPv4, bare IPv6, bracketed IPv6, FQDN), and edge cases (empty/whitespace → None, None passthrough)
- [x] \`TestOperationalVariables::test_syslog_host\` - parser integration test via \`ContextParser.parse()\`
- [x] Full \`just check\` suite: 810 passed, 20 skipped (pre-existing skips), 0 failed

## VyOS command emitted

\`\`\`
set system syslog host <SYSLOG_HOST> facility all level info
\`\`\`

🤖 Generated with [Claude Code](https://claude.com/claude-code) w/ Claude Sonnet 4.6

### CODE REVIEW

File: src/vyos_onecontext/models/config.py [hunk 1]
```diff
@@ -44,6 +44,10 @@ class RouterConfig(BaseModel):
     # Identity
     hostname: Annotated[str | None, Field(None, description="System hostname")]
     ssh_public_key: Annotated[str | None, Field(None, description="SSH public key for vyos user")]
+    syslog_host: Annotated[
+        str | None,
+        Field(None, description="Remote syslog host (IP or hostname) for log forwarding"),
+    ]
```
[the remark this answers is not in the document]
Comment (duplico) [reply src/vyos_onecontext/models/config.py]: Fixed in fae3c45. Added `@field_validator("syslog_host")` to `RouterConfig` that strips whitespace, treats blank-as-None, and restricts values to hostname/IP-safe characters (alphanumeric, dots, hyphens, colons, brackets). Anything with whitespace, quotes, or shell metacharacters is rejected with a clear error.

(AI-generated via Claude Code w/ Claude Sonnet 4.6)

File: src/vyos_onecontext/generators/system.py [hunk 2]
```diff
@@ -114,6 +114,35 @@ def generate(self) -> list[str]:
...
+        or above to the specified remote host via UDP 514.
+
+        Returns:
+            List with syslog command if syslog_host is set, empty list otherwise
+        """
+        if self.syslog_host is None:
+            return []
+
+        return [
+            f"set system syslog host {self.syslog_host} facility all level info"
```
[the remark this answers is not in the document]
Comment (duplico) [reply src/vyos_onecontext/generators/system.py]: Fixed in fae3c45. Added a `strip()` + empty-check guard in `SyslogGenerator.generate()` so the generator cannot emit a malformed `set system syslog host ...` command even if called with an unvalidated value. Model-level validation in `RouterConfig` is the primary defence; the generator guard is a belt-and-suspenders backstop.

(AI-generated via Claude Code w/ Claude Sonnet 4.6)

File: src/vyos_onecontext/models/config.py [hunk 3]
```diff
@@ -44,6 +44,48 @@ class RouterConfig(BaseModel):
...
+
+        v = v.strip()
+        if not v:
+            return None
+
+        # Allow only safe characters: alphanumeric, dots, hyphens, colons (IPv6),
+        # and square brackets (IPv6 with brackets). No whitespace, quotes, or
+        # shell-special characters.
+        pattern = r"^[a-zA-Z0-9.\-:\[\]]+$"
+        if not re.match(pattern, v):
```
[the remark this answers is not in the document]
Comment (duplico) [reply src/vyos_onecontext/models/config.py]: Fixed in 4a4813e. Updated the docstring to accurately describe this as a safe-token check (character allowlist) rather than claiming strict RFC 1123 or IP address validation. Added a note that VyOS will reject invalid values at commit time if they aren't valid hostnames or IPs.

(AI-generated via Claude Code w/ Claude Sonnet 4.6)

File: tests/test_generators.py [hunk 4]
```diff
@@ -4138,3 +4139,69 @@ def test_generate_config_management_and_relay_vrf_ordering(self):
...
+        assert commands[0] == "set system syslog host mon.infra.swccdc.com facility all level info"
+
+    def test_generate_with_ip_address(self):
+        """Test syslog configuration with an IP address."""
+        gen = SyslogGenerator("192.168.1.10")
+        commands = gen.generate()
+
+        assert len(commands) == 1
+        assert commands[0] == "set system syslog host 192.168.1.10 facility all level info"
+
```
[the remark this answers is not in the document]
Comment (duplico) [reply tests/test_generators.py:4162]: Fixed in 4a4813e. Added `test_generate_with_ipv6_bare` and `test_generate_with_ipv6_bracketed` to `TestSyslogGenerator` in test_generators.py, verifying the emitted command string for both bare (`2001:db8::1`) and bracketed (`[2001:db8::1]`) IPv6 addresses.

(AI-generated via Claude Code w/ Claude Sonnet 4.6)

File: src/vyos_onecontext/generators/system.py [hunk 5]
```diff
@@ -114,6 +114,33 @@ def generate(self) -> list[str]:
...
+        Emits a single VyOS command that forwards all facilities at info level
+        or above to the specified remote host via UDP 514.
+
+        Returns:
+            List with syslog command if syslog_host is set, empty list otherwise
+        """
+        if self.syslog_host is None:
+            return []
+
+        return [f"set system syslog host {self.syslog_host} facility all level info"]
```
[the remark this answers is not in the document]
Comment (duplico) [reply src/vyos_onecontext/generators/system.py:141]: Intentional design decision: all generators in this codebase use a `None`-only guard (e.g., `HostnameGenerator`, `SshKeyGenerator`). The model validator already guarantees `syslog_host` is either `None` or a validated non-empty token — empty/whitespace strings are coerced to `None` before the value reaches the generator. Adding a `.strip()` guard here but not in other generators would create an inconsistency in the pattern without meaningful safety benefit.

(AI-generated via Claude Code w/ Claude Opus 4.6)
